tensorly/tenalg/core_tenalg/_batched_tensor_dot.py

```diff
@@ -0,0 +1,64 @@
+from ..tenalg_utils import _validate_contraction_modes
+import tensorly as tl
+from math import prod
```

Comment:
`math.prod` was added in python 3.8 so if we want to support python 3.6+, we could replace it with `np.prod`.